core/src/main/scala/com/microsoft/azure/synapse/ml/fabric/OpenAITokenLibrary.scala

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.azure.synapse.ml.fabric
+
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import org.json.JSONObject
+import pdi.jwt.{Jwt, JwtOptions}
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+
+import java.util.Date
+import scala.util.{Failure, Success, Try}
+
+object OpenAITokenLibrary extends SynapseMLLogging with AuthHeaderProvider {
```

Comment:
I believe we already have a pared down version of this here:

https://github.com/microsoft/SynapseML/blob/9b008b58ae84eb58772c00650492919686ef2c05/core/src/main/scala/com/microsoft/azure/synapse/ml/logging/fabric/CertifiedEventClient.scala#L16

Reply:
I am extracting most of its code as a generic FabricClient and moving them outside of the "logging" folder. And I am modifying the CertifiedEventClient to call FabricClient

Reply:
sounds great